### ISSUE TITLE
Incorrect downloads on Julia master?

### ISSUE THREAD

**Issue by malmaud** [JuliaWeb/Requests.jl#145]
On julia .5:

```julia
using Requests
length(get("https://storage.googleapis.com/malmaud-stuff/tensorflow_linux_cpu_v4.zip").data)
```
17021961 (this is correct)

On Julia master (0.6.0-dev.2319), that gives 17021963.

**Comment by quinnj** [comment]
Hmmm....I actually can't reproduce this in HTTP.jl, 0.5 and 0.6 give 17021961. Is this a consistent result for you? For me, this does take almost 5 minutes to download, even w/ my pretty decent home wifi (through chrome, it takes seconds).

**Comment by malmaud** [comment]
Ya, using `Requests` this is consistent for me (although it also downloads in a few seconds for me, as it does in the browser).

**Comment by tanmaykm** [comment]
Faced this as well.
Null bytes precede the actual data with Julia master.

````
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.0-dev.2278 (2017-01-24 00:34 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 75302ae* (5 days old master)
|__/                   |  x86_64-linux-gnu

julia> using Requests

julia> s = readall(get("http://httpbin.org/get"))
"\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0{\n  \"args\": {}, \n  \"headers\": {\n    \"Accept\": \"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\", \n    \"Host\": \"httpbin.org\", \n    \"User-Agent\": \"Requests.jl/0.0.0\"\n  }, \n  \"origin\": \"122.179.15.152\", \n  \"url\": \"http://httpbin.org/get\"\n}\n"
````

With Julia 0.5:
````
julia> using Requests

julia> s = readall(get("http://httpbin.org/get"))
"{\n  \"args\": {}, \n  \"headers\": {\n    \"Accept\": \"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\", \n    \"Host\": \"httpbin.org\", \n    \"User-Agent\": \"Requests.jl/0.0.0\"\n  }, \n  \"origin\": \"122.179.15.152\", \n  \"url\": \"http://httpbin.org/get\"\n}\n"
````

**Comment by Keno** [comment]
👍  here as well. Will take a look.

**Comment by malmaud** [comment]
Might want to start your search around https://github.com/JuliaWeb/MbedTLS.jl/blob/master/src/ssl.jl#L217

**Comment by Keno** [comment]
The example above by @tanmaykm was http though.

**Comment by malmaud** [comment]
When you're right, you're right.

**Comment by malmaud** [comment]
Closed by #146
